app/src/main/java/com/yml/womensafety/navigationdrawer/home/HomePageActivity.kt

```diff
@@ -1,28 +1,74 @@
 package com.yml.womensafety.navigationdrawer.home
 
+import android.Manifest
+import android.content.ActivityNotFoundException
 import android.content.Intent
+import android.content.pm.PackageManager
+import android.location.Geocoder
+import android.location.Location
 import android.os.Bundle
+import android.telephony.SmsManager
+import android.util.Log
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.ViewModelProvider
 import com.google.android.material.bottomnavigation.BottomNavigationView
-import com.yml.womensafety.FirebaseUtil
-import com.yml.womensafety.FragmentUtil
-import com.yml.womensafety.R
+import com.google.android.material.snackbar.Snackbar
+import com.yml.womensafety.*
+import com.yml.womensafety.AlertDialogUtil.showAlert
 import com.yml.womensafety.authentication.LoginActivity
 import com.yml.womensafety.navigationdrawer.UserProfile
 import com.yml.womensafety.navigationdrawer.contacts.ContactsFragment
+import com.yml.womensafety.viewModel.ContactsViewModel
+import kotlinx.android.synthetic.main.activity_home_page.*
+import java.util.*
+
 
 class HomePageActivity : AppCompatActivity() {
+    companion object {
+        private const val Request_Code_permission: Int = 2
+        private const val LOCATION_PERMISSION_CODE = 1
+        private const val LOG_MESSAGE = "Error"
+        private const val index = 2
```

Comment:
Name the constants in caps separated by underscore

Reply:
Done

Reply:
Done